the-collab-lab/tcl-62-smart-shopping-list
# Style List Page

## Description

This pull request is to include styling within the List page view of the application. It includes changes to `List.jsx`, `ListItem.jsx` and `ListItem.css`

## Related Issue

closes #31 

## Acceptance Criteria
* _NOTE: most of these criteria are incomplete and could use some finishing touches/polishing_
Empty list state:
- [x] Style the empty list state

Non-empty list state:
- [x] Style the search bar
- [x] Style list items so that the names are left aligned, purchase urgency is left aligned, and delete buttons are left aligned
- [x] Style purchase urgency label 
- [x] Remove the ListItem.css file so all styling is done in the List.jsx and ListItem.jsx components via ChakraUI

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|  ✓     | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|   ✓     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

In its current form, the List view has been updated. An empty list has the text centered similarly to the Nav and Header items, and the +Add Item button has received some introductory styling. 

When the List view is populated, the search bar as received styling, as have the List items, and the Urgency Indicator

### Before EMPTY LIST
![listEmptyBefore](https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/92347861/09986ca5-84ee-4ad1-9d49-73d0b94a1d59)

### After EMPTY LIST
![listEmptyAfter](https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/92347861/8950891b-e28b-4b56-993e-067c0b9dbffb)

### Before POPULATED LIST
![listPopBefore](https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/92347861/9b54c274-0500-4745-914c-4bc5a5cc2e6a)

### After POPULAED LIST
![listPopAfter](https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/92347861/0e4e3347-5b43-44c6-8e26-e2520afa1e37)

## Testing Steps / QA Criteria

1. From your main branch gather all changes by running `git fetch` and `git pull` in your terminal
2. Make sure you have all dependencies updated by running `npm i` or `npm install`
3. Switch to feature branch `style-list-page`
4. Start with an empty list. If there is a list token active in your browser, open the devtools and navigate to the Application Tab and delete your three-word token
5. Navigate back to the app, and start adding items
6. For each item added, indicate the urgency with which they will be purchased
7. When finished, delete your new list from the Firebase console to keep the database tidy

## TO-DOs

- [x] Move all styling from ListItem.css to ListItem.jsx
- [x] Fix the layout of the populated List within List.jsx to be aligned with the search bar
- [x] Adjust alignment of elements within each List Item
